homeassistant/components/whirlpool/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Whirlpool Sixth Sense integration."""
+import logging
+
+import aiohttp
+from whirlpool.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Whirlpool Sixth Sense from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    auth = Auth(entry.data["username"], entry.data["password"])
+    try:
+        await auth.do_auth(store=False)
+    except aiohttp.ClientError as ex:
+        _LOGGER.error("Connection error")
+        raise ConfigEntryNotReady("Cannot connect") from ex
+
+    if not auth.is_access_token_valid():
+        _LOGGER.error("Authentication failed")
+        raise ConfigEntryNotReady("Authentication failed")
```

Comment:
```suggestion
        return False
```

If you `ConfigEntryNotReady` the system will retry later. Since the token is not valid this will result hammering the api as it will try to login over and over again.